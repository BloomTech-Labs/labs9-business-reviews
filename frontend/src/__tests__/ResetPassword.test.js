import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import ResetPasswordComponent from '../components/ResetPassword';

describe('ResetPasswordComponent />', () => {
  it('renders', () => {
    shallow(<ResetPasswordComponent />);
  });
  it('matches snapshot', () => {
    const wrapper = shallow(<ResetPasswordComponent />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should contain NavBar', () => {
    let app = shallow(<ResetPasswordComponent/>);
    expect(app.find("NavBar").length).toBe(1);
  });
  it('should contain one h1 with text "Forgot your password?"', () => {
    let app = shallow(<ResetPasswordComponent/>);
    expect(app.find("h1").length).toBe(1);
    expect(app.find("h1").text()).toMatch("Forgot your password?");
  });
  it('should contain one form element', () => {
    let app = shallow(<ResetPasswordComponent/>);
    expect(app.find("form").length).toBe(1);
  });
  it('should contain one label', () => {
    let app = shallow(<ResetPasswordComponent/>);
    expect(app.find("label").length).toBe(1);
  });
  it('should contain two input elements', () => {
    let app = shallow(<ResetPasswordComponent/>);
    expect(app.find("input").length).toBe(2);
  });
});
