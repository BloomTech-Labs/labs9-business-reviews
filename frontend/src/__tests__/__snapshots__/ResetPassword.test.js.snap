// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordComponent /> matches snapshot 1`] = `
<div>
  <NavBar />
  <styled.div>
    <h1>
      Forgot your password?
    </h1>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="email"
      >
        Check Email:
      </label>
      <input
        name="email"
        onChange={[Function]}
        placeholder="Check your email see if it exists"
        type="text"
        value=""
      />
      <input
        type="submit"
        value="âž¤"
      />
    </form>
  </styled.div>
</div>
`;
